src/commands/push-branches.js

```diff
@@ -13,24 +13,36 @@ exports.handler = async (_, state) =>
 			})
 			const repoBranch = incrementSuffix(branches, repo.localBranch)
 
-			await logger.logPromise(
-				git.push({
-					repository: 'origin',
-					refspec: `${repo.localBranch}:${repoBranch}`,
-					workingDirectory: repo.clone,
-				}),
-				`pushing ${styles.branch(repo.localBranch)} to ${styles.repo(
-					`${repo.owner}/${repo.name}`,
-				)}${
-					repoBranch !== repo.localBranch
-						? ` (as ${styles.branch(repoBranch)} because ${styles.branch(
-								repo.localBranch,
-						  )} already exists)`
-						: ''
-				}`,
-			)
+			const newCommits = await git.cherry({
+				workingDirectory: repo.clone,
+				origin: 'origin',
+				branch: repo.localBranch,
+			})
 
-			repo.remoteBranch = repoBranch
+			if (newCommits.length > 0) {
+				await logger.logPromise(
+					git.push({
+						repository: 'origin',
+						refspec: `${repo.localBranch}:${repoBranch}`,
+						workingDirectory: repo.clone,
+					}),
+					`pushing ${styles.branch(repo.localBranch)} to ${styles.repo(
+						`${repo.owner}/${repo.name}`,
+					)}${
+						repoBranch !== repo.localBranch
+							? ` (as ${styles.branch(repoBranch)} because ${styles.branch(
+									repo.localBranch,
+							  )} already exists)`
+							: ''
+					}`,
+				)
+				repo.remoteBranch = repoBranch
+			} else {
+				logger.log('Skipping branch push', {
+					status: 'info',
+					message: `No commits have been made on the '${repoBranch}' branch, so pushing it has been skipped.`,
```

Comment:
this message should include the name of the repo, otherwise you can't tell what's been skipped:

<img width="577" alt="image" src="https://user-images.githubusercontent.com/631757/93354249-b365c300-f834-11ea-8b71-ff6ce44a81b9.png">

(it's fine for now but it would also be good to remove that whitespace between the logs but i don't know where it comes from so 🤷‍♀️)

Reply:
the whitespace is coming from the module `komatsu`. It gets added whenever you add a status that is not the default spinner :/

Reply:
ahhhh yes okay. it inserts whitespace if you log something with an id that doesn't have a currently in-progress spinner. i think the only way around this would be to not use `logPromise` but do custom logging with the same id throughout the handler, like [`create-project` does it](https://github.com/Financial-Times/nori/blob/25d85c664f44ab68b6781cfa986bb368d5e17d3f/src/commands/create-project.js#L37). think it's fine for now though